test/Test/Database/LSMTree/Normal/StateMachine.hs

```diff
@@ -393,8 +394,8 @@ instance ( Eq (Class.TableConfig h)
   x == y = go x y
     where
       go :: LockstepAction (ModelState h) a -> LockstepAction (ModelState h) a -> Bool
-      go (New conf1)                (New conf2) =
-          conf1 == conf2
+      go (New p1 conf1)                (New p2 conf2) =
+          Just p1 == cast p2 && conf1 == conf2
```

Comment:
`p1 == p2` doesn't do anything. Proxies are always equal.

Reply:
I'm casting beforehand 
```
ghci> cast (Proxy @Int) == Just (Proxy @Char)
False
ghci> cast (Proxy @Int) == Just (Proxy @Int)
True
```

Reply:
The cast adds type-check which didn't exist before. It's cast which fails, not the equality.

Reply:
To clarify: there weren't type-equality check before, so this is semantic change, not a compatibility change.

Reply:
Okay, I get your point -- I'm now using `eqT` instead and checking that the result of `Just Refl`. The test looks like it can't fail because the type of `New` mentions `k`, `v` and `blob`, but it doesn't hurt to keep it in there